interview-questions.md

```diff
@@ -8,62 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: The predefined parameters that the .map() method can take are: value, index, array. The value is required and the other two are optional. 
 
-  Researched answer:
+  Researched answer: Value, index, and array are the predefined parameters that the .map() method can take. Value is the required parameter while index and array are optional. Since the .map() method iterates through each value at each index in the given array, it is the most important parameter. The method returns a new array based on the values at each index in the input array.
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: Map iterates through a given array and returns a new array with the same number of indexes as the given array. Filter iterates through a given array and returns a new array with the specified values taken out of the the given array.
 
-  Researched answer:
+  Researched answer: Filter iterates through a given array, takes out prescribed items in the array, and places them into a new array. The length of this new array is less than the original array due to the nature of filtering--cherry picking--certain values out of that original array. Map iterates through a given array, modifies the values at each index of the array, and returns a new array with the modified values that is the same length as the input array.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: Functions are a set of instructions that return an output that is based on an input that is passed as an argument in its parameter. A method is similar in that it contains a set of an instructions to produce an output, but a method is usually associated with an object.
 
-  Researched answer:
+  Researched answer: Functions are a set of instructions that produce a desired output based on data that has been inputted into the fucntion. Functions can stand on their own within the code, while methods are functions associated with objects. A function can be called by invoking its name; a method must be attached to the name of its associated object using dot notation or bracket notation.
```

Comment:
Great explanation of the ideas! Good job calling out the use case-- next week take it a step further and share a time you personally used the tools in question.